src/gui/systray.cpp

```diff
@@ -224,6 +224,31 @@ bool Systray::useNormalWindow() const
     return cfg.showMainDialogAsNormalWindow();
 }
 
+bool Systray::darkMode()
+{
+    if(!_guiAppInstance) {
+        QGuiApplication *ptr = qobject_cast<QGuiApplication *>(QGuiApplication::instance());
+        if(ptr) {
+            _guiAppInstance.reset(ptr);
+            connect(ptr, &QGuiApplication::paletteChanged, this, &Systray::darkModeChanged);
+        }
+    }
```

Comment:
same question, why do you not connect in the `Systray` constructor ?
and I know our code in this part is a real mess

Reply:
Like with Theme, the ptr is a nullptr when created in the constructor

Reply:
Ok unlike with the Theme class there is no issue with putting this in the constructor, so will move it there